File: lab_5_cs/lab_5_cs/TCylinder.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace lab_5_cs
{
    class TCylinder : TFigure
    {
        private double _radius;
        private double _height;

        public TCylinder(double radius, double heigh)
        {
            _radius = radius;
            _height = heigh;
        }
        
        public override double FindSurfaceArea()
        {
            double sideSurfaceArea = 2 * Math.PI * _radius * _height;
            double baseArea = Math.PI * Math.Pow(_radius, 2);
            return sideSurfaceArea + baseArea;
        }

        public override double FindVolume()
        {
            return Math.PI * Math.Pow(_radius, 2) * _height;
        }
    }
}

File: lab_5/lab_5_py/lab_5_py/TCylinder.py
from TFigure import *

class TCylinder(TFigure):
    
    __radius = 0.0

    def __init__(self, cylinder):
        self.__radius = float(cylinder[0])
        self._height = float(cylinder[1])

    def find_surface_area(self):
        side_surface_area = 2 * self._Pi * self.__radius * self._height
        base_area = self._Pi * pow(self.__radius, 2)
        return side_surface_area + base_area

    def find_volume(self):
        return self._Pi * pow(self.__radius, 2) * self._height